cudamapper/include/claragenomics/cudamapper/overlapper.hpp

```diff
@@ -52,6 +48,14 @@ class Overlapper
                                 const std::vector<Overlap>& overlaps,
                                 size_t min_residues    = 20,
                                 size_t min_overlap_len = 50);
+
+    /// \brief updates read names for vector of overlaps output from get_overlaps
+    /// \param overlaps input vector of overlaps generated in get_overlaps
+    /// \param index_query
+    /// \param index_target
+    static void update_read_names(std::vector<Overlap>& overlaps,
+                                  const Index& index_query,
```

Comment:
I have a more general and a bit controversial suggestion. `update_read_names()` actually only requires `read_id_to_read_name()` and `read_id_to_read_length()`, which are both host arrays. As `update_read_names()` is supposed to run in parallel with another anchor generating iteration this means that these indices would take up (scarce) device memory.
My suggestion: only pass `read_id_to_read_name` and `read_id_to_read_length` vectors for both indices. That way the user (author of `main.cu`) can decide if they want to pay the price of making copies of those vectors in order to save some device space or they are fine with keeping the whole `Index` object alive (which might make sense in device index caching scenario).
@vellamike for opinion

Reply:
@mimaric I see two sides to this:

1. It is cleaner how it is at the moment, the `Index` will only in practice live for a very short extra time anyway
2. In principle, this not only wastes device memory but also can delay eviction from device cache, leading to confusing behaviour. 

My gut instinct is that the extra H2H copy will be expensive and the code a little uglier so we shouldn't do your suggestion. What do you think?

Reply:
I agree. Let's keep it as it is for now and do some more analysis afterwards. That's why I said it's a bit controversial.

Reply:
+1

Reply:
left as it is.